### ISSUE TITLE
Feature request: specify new session name as modules option

### ISSUE THREAD

**Issue by ivonutar** [rapid7/metasploit-framework#11633]
It would be a nice feature, specifically for us trying to automate things, to specify the name of session-to-be-created before the actual module is executed. Currently it is only possible to rename an already existing session.

I can't think of any reliable way of how to fast find out, if the module execution (using RPC) was successful or not - there is no return value, '[+]' and '[-]' can be misleading and the only way to see, if execution was successful is via checking the newly created session. This is easier when doing things manually, but inside a script might not be doable. By naming the session beforehand, it would be really easy to just check if the session with that name appeared.

**Comment by wvu-r7** [comment]
#8821

**Comment by busterb** [comment]
Hmm, almost like you need to be able to check the value of `$?`. I think having the new session from an exploit have a particular name isn't too hard, but it sounds like more fundamentally you just want a success/failure for module runs via RPC. What method are you using to run modules currently?

**Comment by ivonutar** [comment]
Exactly, something like `$?`would be one possible solution. I am currently using pymetasploit (for python) that implements API for RPC. You have two options on how tu run modules:
1. Using "module" endpoint of RPC directly. This way it is not possibe to read the stdout, which is a problem for me.
2. Using "console" endpoint and writing all commands as a string. This way I can get stdout of module execution, but checking the success is more difficult.

I am using option 2. My method of checking success currently is to
1. before execution get the list of open sessions
2. before execution get the list of running jobs
3. execute using "execute -j"
4. wait 1 second so the job has a chance of being created (this is not 100%  reliable)
5. In loop check the "busy" parameter of the client (maybe its a python api thing, so not really msf issue)
6. After finishing (busy=false) check "OptionValidationError", "Success" and other possible strings in stdout.
7. Most importantly, get the list of all open sessions, subtract all, that were open before execution, and check the newly created ones, that have a target set to the target of the script.

This currently works for me, but as I said, it is not really reliable. It would be nice to have some feedback from RPC, maybe a having a stdout/stderr as a return value for "module.execute" or "job.info" endpoint.
